core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/DecorrelateLeftUnnestWithGlobalAggregation.java

```diff
@@ -0,0 +1,277 @@
+/*
```

Comment:
How much common code is here and in https://github.com/trinodb/trino/pull/6326. Should those be merged maybe?

Reply:
They only share the tiniest helper methods. The rest of the code looks similar but differs in details. I'd rather not merge them because the code might be hard to follow. Nor put them in a single file.